Sources/Templates/AutoEquatable.stencil

```diff
@@ -0,0 +1,64 @@
+import Danger
+
+// swiftlint:disable file_length
+fileprivate func compareOptionals<T>(lhs: T?, rhs: T?, compare: (_ lhs: T, _ rhs: T) -> Bool) -> Bool {
+    switch (lhs, rhs) {
+    case let (lValue?, rValue?):
+        return compare(lValue, rValue)
+    case (nil, nil):
+        return true
+    default:
+        return false
+    }
+}
+
+fileprivate func compareArrays<T>(lhs: [T], rhs: [T], compare: (_ lhs: T, _ rhs: T) -> Bool) -> Bool {
+    guard lhs.count == rhs.count else { return false }
+    for (idx, lhsItem) in lhs.enumerated() {
+        guard compare(lhsItem, rhs[idx]) else { return false }
+    }
+
+    return true
+}
+
+{% macro compareVariables variables %}
+    {% for variable in variables where variable.readAccess != "private" and variable.readAccess != "fileprivate" %}{% if not variable.annotations.skipEquality %}guard {% if not variable.isOptional %}{% if not variable.annotations.arrayEquality %}lhs.{{ variable.name }} == rhs.{{ variable.name }}{% else %}compareArrays(lhs: lhs.{{ variable.name }}, rhs: rhs.{{ variable.name }}, compare: ==){% endif %}{% else %}compareOptionals(lhs: lhs.{{ variable.name }}, rhs: rhs.{{ variable.name }}, compare: ==){% endif %} else { return false }{% endif %}
+    {% endfor %}
+{% endmacro %}
+
+// MARK: - AutoEquatable for classes, protocols, structs
+{% for type in types.implementing.AutoEquatable|!enum %}
+// MARK: - {{ type.name }} AutoEquatable
+{% if not type.kind == "protocol" %}extension {{ type.name }}: Equatable {}{% endif %}
+{% if type.supertype.based.Equatable or type.supertype.implements.AutoEquatable %}THIS WONT COMPILE, WE DONT SUPPORT INHERITANCE for AutoEquatable{% endif %}
+{{ type.accessLevel }} func == (lhs: {{ type.name }}, rhs: {{ type.name }}) -> Bool {
+    {% if not type.kind == "protocol" %}
+    {% call compareVariables type.storedVariables %}
+    {% else %}
+    {% call compareVariables type.allVariables %}
+    {% endif %}
+    return true
+}
+{% endfor %}
+
+// MARK: - AutoEquatable for Enums
+{% for type in types.implementing.AutoEquatable|enum %}
+// MARK: - {{ type.name }} AutoEquatable
+extension {{ type.name }}: Equatable {}
+{{ type.accessLevel }} func == (lhs: {{ type.name }}, rhs: {{ type.name }}) -> Bool {
+    switch (lhs, rhs) {
+    {% for case in type.cases %}
+    {% if case.hasAssociatedValue %}case (.{{ case.name }}(let lhs), .{{ case.name }}(let rhs)):{% else %}case (.{{ case.name }}, .{{ case.name }}):{% endif %}
+        {% ifnot case.hasAssociatedValue %}return true{% else %}
+        {% if case.associatedValues.count == 1 %}
+        return lhs == rhs
+        {% else %}
+        {% for associated in case.associatedValues %}if lhs.{{ associated.externalName }} != rhs.{{ associated.externalName }} { return false }
+        {% endfor %}return true
+        {% endif %}
+        {% endif %}
+    {% endfor %}
+    {% if type.cases.count > 1 %}default: return false{% endif %}
+    }
+}
+{% endfor %}
```

Comment:
Can you please add docs (the readme or contributing) somewhere on how to update these, I've never used it

Reply:
Sure thing! (Sourcery is available on Homebrew, but I kind of wish there was a way to include it in the project directory like with a Gemfile....)

Reply:
Added in this commit: 66ad8fccaeb3965384f21207fb8da53aa1550e8f